src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -154,64 +178,53 @@ public void simulationPeriodic() {
     }
   }
 
-  // Spins up the shooter, and requests feeding it when the rollers are within parameters.
-  // Receives distance-to-target from Limelight, or other sensor.
-  // Required to be called repeatedly; consider pub-sub for LimelightGetDistance() or equivalent
-  // method to save a method call
+  // Spins up the shooter. Receives distance-to-target from vision.
   public Command revShooter() {
-
-    // Only runs if the exit code from the limelight status function returns 0!
-
     return Commands.startEnd(
         () ->
-            this.setShooterSpeeds(
-                new SpeakerConfig(-1, targetRPM.getDouble(0), rightTargetRPM.getDouble(0))),
+            this.setShooterSpeeds( // Should receive distance-to-target from vision
+                new SpeakerConfig(
+                    -1, leftShooterRPM.getDouble(4000), rightShooterRPM.getDouble(4000))),
         () -> {
           shooterLeftMotor.stopMotor();
           shooterRightMotor.stopMotor();
         },
-        this); // TODO: Replace distance and exit code with LimelightGetDistance() and
-    // CheckLimelightStatus() respectively
+        this);
   }
 
   public Command bumperShoot() {
-    return run(() -> {
-          setShooterSpeeds(speakerConfigList[0]);
-        })
+    return startEnd(
+            () -> {
+              setShooterSpeeds(speakerConfigList[0]);
+            },
+            () -> {
+              shooterLeftMotor.stopMotor();
+              shooterRightMotor.stopMotor();
+              leftMotorTargetSpeedEntry.log(0d);
+              rightMotorTargetSpeedEntry.log(0d);
```

Comment:
Also add the entry logging to revShooter Command. Better yet, make a private method to stop the motors and log the entries, then just plug that method in to both Commands' end Runnables.

Reply:
Resolved in b5ecadd